dtrace-parser/src/lib.rs

```diff
@@ -110,6 +115,93 @@ impl Integer {
         format!("{prefix}int{}_t", self.width_to_c_str())
     }
 
+    pub fn to_asm_op(&self, i: u8) -> String {
+        // See common.rs for note on argument passing and maximum supported
+        // argument count.
+        #[cfg(target_arch = "x86_64")]
+        if cfg!(target_arch = "x86_64") {
+            let base = match i {
+                0 => "di",
+                1 => "si",
+                2 => "d",
+                3 => "c",
+                4 => "8",
+                5 => "9",
+                _ => unreachable!("Up to 6 probe arguments are currently supported"),
+            };
+            let prefix = match self.width {
+                BitWidth::Bit8 => "",
+                BitWidth::Bit16 => "",
+                BitWidth::Bit32 => "e",
+                BitWidth::Bit64 => "r",
+                #[cfg(target_pointer_width = "32")]
+                BitWidth::Pointer => "e",
+                #[cfg(target_pointer_width = "64")]
+                BitWidth::Pointer => "r",
+                #[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
+                BitWidth::Pointer => compile_error!("Unsupported pointer width"),
+            };
+            let suffix = match self.width {
```

Comment:
I think it would be more clear to merge the matches for computing both `prefix` and `suffix`. Right now, the conditional logic on L147 is a bit obtuse, and I think the duplication is worth being very explicit about what prefix / suffix is used when.

Reply:
Merged into a relative huge match statement. Definitely clearer that way: I found that I had bugged the r8 and r9 1 byte, 2 byte and 3 byte names.